packages/react/src/ReactElementValidator.js

```diff
@@ -365,13 +369,20 @@ export function jsxWithValidation(
     }
   }
 
-  if (hasOwnProperty.call(props, 'key')) {
-    if (__DEV__) {
-      console.error(
-        'React.jsx: Spreading a key to JSX is a deprecated pattern. ' +
-          'Explicitly pass a key after spreading props in your JSX call. ' +
-          'E.g. <ComponentName {...props} key={key} />',
-      );
+  if (__DEV__) {
+    if (warnAboutSpreadingKeyToJSX) {
+      if (
+        didWarnAboutSpreadingKey[name] !== true &&
```

Comment:
oh YIKES, lint or flow didn't catch this undefined var (because `window.name` is an actual thing?) 😱

Reply:
followup - https://github.com/facebook/react/pull/18076